src/assets/scripts/client/aircraft/AircraftCommander.js

```diff
@@ -0,0 +1,1093 @@
+                say: 'rerouting as requested'
+            };
+
+            return ['ok', readback];
+        }
+
+        const readback = {
+            log: `your route "${data}" is invalid!`,
+            say: 'that route is invalid!'
+        };
+
+        return ['fail', readback];
+    }
+
+    /**
+      * Removes all legs, and replaces them with the specified route
+      * Note: Input data needs to be provided with single dots connecting all
+      * procedurally-linked points (eg KSFO.OFFSH9.SXC or SGD.V87.MOVER), and
+      * all other points that will be simply a fix direct to another fix need
+      * to be connected with double-dots (eg HLI..SQS..BERRA..JAN..KJAN)
+      *
+      * @for AircraftCommander
+      * @method runReroute
+      * @param data
+      */
+    runReroute(aircraft, data) {
+        // TODO: capitalize everything?
+        data = data[0].toUpperCase();
+        let worked = true;
+        const route = aircraft.fms.formatRoute(data);
+
+        if (worked && route) {
+            // Reset fms
+            worked = aircraft.fms.customRoute(route, true);
+        }
+
+        // TODO: what exactly are we checking here?
+        if (!route || !data || data.indexOf(' ') > -1) {
+            worked = false;
+        }
+
+        // Build the response
+        if (worked) {
+            const readback = {
+                log: `rerouting to: ${aircraft.fms.fp.route.join(' ')}`,
+                say: 'rerouting as requested'
+            };
+
+            return ['ok', readback];
+        }
+
+        const readback = {
+            log: `your route "${data}" is invalid!`,
+            say: 'that route is invalid!'
+        };
+
+        return ['fail', readback];
+    }
+
+    /**
+     * @for AircraftCommander
+     * @method runTaxi
+     * @param data
+     */
+    runTaxi(aircraft, data) {
+        // TODO: all this if logic should be simplified or abstracted
+        if (aircraft.category !== FLIGHT_CATEGORY.DEPARTURE) {
+            return ['fail', 'inbound'];
+        }
+
+        if (aircraft.mode === FLIGHT_MODES.TAXI) {
+            return ['fail', `already taxiing to ${radio_runway(aircraft.rwy_dep)}`];
+        }
+
+        if (aircraft.mode === FLIGHT_MODES.WAITING) {
+            return ['fail', 'already waiting'];
+        }
+
+        if (aircraft.mode !== FLIGHT_MODES.APRON) {
+            return ['fail', 'wrong mode'];
+        }
+
+        // Set the runway to taxi to
+        if (data[0]) {
+            if (this._airportController.airport_get().getRunway(data[0].toUpperCase())) {
+                aircraft.setDepartureRunway(data[0].toUpperCase());
+            } else {
+                return ['fail', `no runway ${data[0].toUpperCase()}`];
+            }
+        }
+
+        // Start the taxi
+        aircraft.taxi_start = this._gameController.game_time();
+        const runway = this._airportController.airport_get().getRunway(aircraft.rwy_dep);
+
+        runway.addQueue(this);
```

Comment:
@Dabea This one is my fault :-(

This line prevents the aircraft info box from appearing when an aircraft has finished taxiing and is first inline to takeoff.

Having abstracted these commands away from the `AircraftInstanceModel`, why is this line now incorrect?  What should be done to fix it?